src/views/App.vue

```diff
@@ -200,36 +200,31 @@ export default {
     ...mapState("userSettings", ["locale"]),
   },
   methods: {
-    loadData() {
+    async loadData() {
       this.loading = true;
-      Promise.all([
-        this.$store.dispatch("auth/index"),
-        this.$store.dispatch("albums/index"),
-        this.$store.dispatch("artists/index"),
-        this.$store.dispatch("genres/index"),
-        this.$store.dispatch("labels/index"),
-        this.$store.dispatch("tracks/index"),
-        this.$store.dispatch("users/index").then(() => {
-          const promises = [];
-          if (this.isModerator) {
-            promises.push(this.$store.dispatch("rescan/show"));
-            promises.push(this.$store.dispatch("codecs/index"));
-            promises.push(this.$store.dispatch("codecConversions/index"));
-            promises.push(this.$store.dispatch("coverFilenames/index"));
-            promises.push(this.$store.dispatch("imageTypes/index"));
-            promises.push(this.$store.dispatch("locations/index"));
-          }
-          return Promise.all(promises);
-        }),
-        new Promise((resolve) => setTimeout(resolve, 1000)),
-      ]).finally(() => {
-        this.loading = false;
-      });
+      const auth = this.$store.dispatch("auth/index");
+      const albums = this.$store.dispatch("albums/index");
+      const artists = this.$store.dispatch("artists/index");
+      const genres = this.$store.dispatch("genres/index");
+      const labels = this.$store.dispatch("labels/index");
+      const tracks = this.$store.dispatch("tracks/index");
+      const users = this.$store.dispatch("users/index");
+      const promises = [auth, albums, artists, genres, labels, tracks, users];
+      await users;
+      if (this.isModerator) {
+        promises.push(this.$store.dispatch("rescan/show"));
+        promises.push(this.$store.dispatch("codecs/index"));
+        promises.push(this.$store.dispatch("codecConversions/index"));
+        promises.push(this.$store.dispatch("coverFilenames/index"));
+        promises.push(this.$store.dispatch("imageTypes/index"));
+        promises.push(this.$store.dispatch("locations/index"));
+      }
+      await Promise.all(promises);
+      this.loading = false;
```

Comment:
This is a semantic change. If things go wrong in the above promises, `this.loading = false` will not be executed. Should be wrapped in a `try { ... } finally { ... }`.